module.py

```diff
@@ -31,61 +27,80 @@ def start(self):
-    log = logging.getLogger()
-
-    # load new modules
-    for file in new_files:
-        try:
-            module = Module(file)
-            modules[module.name] = module
-            new_modules.append(module)
-            log.info("imported '%s' module", module.name)
-        except Exception as e:
-            log.error("importing of '%s' module failed: %s", get_name(file), e)
-            log.debug('%s', traceback.format_exc())
-
-    # get handler dispatcher from core module
-    dispatcher = modules['core'].module.dispatcher
-    def add_handlers(handlers):
-        for handler in handlers:
-            dispatcher.add_handler(handler)
-    def remove_handlers(handlers):
-        for handler in handlers:
-            dispatcher.remove_handler(handler)
-
-    # reload existing modules
-    for module in (m for m in existing_modules if sha256sum(m.path) != m.hashsum):
-        remove_handlers(module.handlers)
-        module.reload()
-        new_modules.insert(0, module)
-        log.info("reloaded '%s' module", module.name)
-
-    # start modules
-    for module in new_modules:
-        module.start()
-        add_handlers(module.handlers)
-        log.info("started '%s' module", module.name)
-
-    # remove old modules
-    for module in old_modules:
-        remove_handlers(module.handlers)
-        del modules[module.name]
-        log.info("removed '%s' module", module.name)
+class ModulesManager:
+    def __init__(self, path):
+        self.path = path
+        self.log = logging.getLogger()
+        self.modules = {}
+        self.new_modules = []
+
+
+    def import_new_modules(self):
+        files = set(glob.glob(self.path))
+        loaded_files = set(m.path for m in self.modules.values())
+        new_files = files - loaded_files
+
+        # load new modules
+        for file in new_files:
+            try:
+                module = Module(file)
+                self.modules[module.name] = module
+                self.new_modules.append(module)
+                self.log.info("imported '%s' module", module.name)
+            except Exception as e:
+                self.log.error("importing of '%s' module failed: %s", get_name(file), e)
+                self.log.debug('%s', traceback.format_exc())
+
+
+    def start_modules(self, dispatcher):
+        self.dispatcher = dispatcher
+
+        for module in self.new_modules:
+            module.start()
+            for handler in module.handlers:
+                self.dispatcher.add_handler(handler)
+            self.log.info("started '%s' module", module.name)
+
+        self.new_modules.clear()
+
+
+    def reload_modules(self):
+        # helper function
+        def remove_handlers(handlers):
+            for handler in handlers:
+                self.dispatcher.remove_handler(handler)
+
+        files = set(glob.glob(self.path))
+        loaded_files = set(m.path for m in self.modules.values())
+        old_modules = [self.modules[get_name(p)] for p in loaded_files - files]
```

Comment:
(optional) It's quite tricky to read. Please add comment above with variable description.